test/metabase/query_processor/middleware/metrics_test.clj

```diff
@@ -60,6 +61,72 @@
 (def adjust
   (comp #'metrics/adjust #'fetch-source-query/resolve-source-cards))
 
+(defn- check-prometheus-metrics!
+  [& {expected-metrics-count  :metabase-query-processor/metrics
+      expected-metrics-errors :metabase-query-processor/metrics-errors
+      metric-and-mp           :metric-and-mp
+      query-fn                :query-fn
+      check-fn                :check-fn}]
+  (let [[source-metric mp] (or metric-and-mp (mock-metric))
+        query              (if query-fn
+                             (query-fn mp source-metric)
+                             (lib/query mp source-metric))
+        calls              (atom nil)]
+    (with-redefs [prometheus/inc! #(swap! calls conj %)]
```

Comment:
Why not count directly with sg. like
```suggestion
    (with-redefs [prometheus/inc! #(swap! calls update % (fnil inc 0))]
```

Reply:
Tbh I copied this idiom from the existing prometheus-metrics-counting tests in `metabase.email-test/send-message!-test`, but agree that counting directly seems better.